diff --git a/android/build.gradle b/android/build.gradle
index 0000000..1111111 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -106,7 +106,9 @@ repositories {
 dependencies {
   // implementation "com.facebook.react:react-android"
   // api "com.facebook.react:react-native:+"
-  implementation "com.facebook.react:react-android:0.78.2"
+  implementation "com.facebook.react:react-native:+"
+  implementation "androidx.fragment:fragment:1.3.6"
+  implementation "androidx.fragment:fragment-ktx:1.3.6"

   def kotlin_version = getExtOrDefault("kotlinVersion")
   implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
