diff --git a/android/src/main/java/com/appsflyer/reactnative/RNAppsFlyerModule.java b/android/src/main/java/com/appsflyer/reactnative/RNAppsFlyerModule.java
index bb77d9363d132be9ff5b47541820384e23d7f2a1..c97c2d7bc6a04e1630f5b4aee41aed3fe516239e 100755
--- a/android/src/main/java/com/appsflyer/reactnative/RNAppsFlyerModule.java
+++ b/android/src/main/java/com/appsflyer/reactnative/RNAppsFlyerModule.java
@@ -54,6 +54,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Iterator;
 import java.util.Locale;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 import static com.appsflyer.reactnative.RNAppsFlyerConstants.*;
 import static com.appsflyer.reactnative.RNAppsFlyerConstants.afOnDeepLinking;
@@ -323,14 +324,20 @@ public class RNAppsFlyerModule extends ReactContextBaseJavaModule {
             Activity currentActivity = getCurrentActivity();
             if (currentActivity != null) {
                 AppsFlyerLib.getInstance().logEvent(getCurrentActivity(), eventName, data, new AppsFlyerRequestListener() {
+                    final AtomicBoolean isCompleted = new AtomicBoolean(false);
+
                     @Override
                     public void onSuccess() {
-                        successCallback.invoke(SUCCESS);
+                        if (isCompleted.compareAndSet(false, true)) {
+                            successCallback.invoke(SUCCESS);
+                        }
                     }
 
                     @Override
                     public void onError(int i, @NonNull String s) {
-                        errorCallback.invoke(s);
+                        if (isCompleted.compareAndSet(false, true)) {
+                            errorCallback.invoke(String.valueOf(i), s);
+                        }
                     }
                 });
             }
@@ -355,14 +362,20 @@ public class RNAppsFlyerModule extends ReactContextBaseJavaModule {
             Activity currentActivity = getCurrentActivity();
             if (currentActivity != null) {
                 AppsFlyerLib.getInstance().logEvent(getCurrentActivity(), eventName, data, new AppsFlyerRequestListener() {
+                    final AtomicBoolean isCompleted = new AtomicBoolean(false);
+
                     @Override
                     public void onSuccess() {
-                        promise.resolve(SUCCESS);
+                        if (isCompleted.compareAndSet(false, true)) {
+                            promise.resolve(SUCCESS);
+                        }
                     }
 
                     @Override
                     public void onError(int i, @NonNull String s) {
-                        promise.reject(s);
+                        if (isCompleted.compareAndSet(false, true)) {
+                            promise.reject(String.valueOf(i), s);
+                        }
                     }
                 });
             }
